tstoremem/metadata.go

```diff
@@ -29,7 +29,39 @@ func NewThreadMetadata() tstore.ThreadMetadata {
 	}
 }
 
-func (ts *memoryThreadMetadata) PutMeta(t thread.ID, key string, val interface{}) error {
+func (ts *memoryThreadMetadata) PutInt64(t thread.ID, key string, val int64) error {
+	ts.putValue(t, key, val)
+	return nil
+}
+
+func (ts *memoryThreadMetadata) GetInt64(t thread.ID, key string) (*int64, error) {
+	val := ts.getValue(t, key).(int64)
+	return &val, nil
+}
+
+func (ts *memoryThreadMetadata) PutString(t thread.ID, key string, val string) error {
+	ts.putValue(t, key, val)
+	return nil
+}
+
+func (ts *memoryThreadMetadata) GetString(t thread.ID, key string) (*string, error) {
+	val := ts.getValue(t, key).(string)
+	return &val, nil
+}
+
+func (ts *memoryThreadMetadata) PutBytes(t thread.ID, key string, val []byte) error {
+	b := make([]byte, len(val))
+	copy(b, val)
```

Comment:
Changed here to copy the byte slice. Might be debatable though, but I think is safer than a further change in the slice after `PutBytes` doesn't also mean changing the stored value.